smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/WaiterGenerator.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.kotlin.codegen.rendering.waiters
+
+import software.amazon.smithy.kotlin.codegen.core.KotlinWriter
+import software.amazon.smithy.kotlin.codegen.core.RuntimeTypes
+import software.amazon.smithy.kotlin.codegen.core.withBlock
+import java.text.DecimalFormat
+import java.text.DecimalFormatSymbols
+
+/**
+ * Renders the top-level retry strategy for a waiter.
+ */
+private fun KotlinWriter.renderRetryStrategy(wi: WaiterInfo) {
+    setOf(
+        RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitterOptions,
+        RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitter,
+        RuntimeTypes.Core.Retries.Impl.StandardRetryStrategyOptions,
+        RuntimeTypes.Core.Retries.Impl.StandardRetryStrategy,
+        RuntimeTypes.Core.Retries.Impl.NoOpTokenBucket,
+    ).forEach(::addImport)
+
+    write("")
+    withBlock("private val #L = run {", "}", wi.retryStrategyName) {
+        withBlock("val delayOptions = ExponentialBackoffWithJitterOptions(", ")") {
+            write("initialDelayMs = #L,", wi.waiter.minDelay.msFormat)
+            write("scaleFactor = 1.5,")
+            write("jitter = 1.0,")
+            write("maxBackoffMs = #L,", wi.waiter.maxDelay.msFormat)
+        }
+        write("val delay = ExponentialBackoffWithJitter(delayOptions)")
+        write("")
+        write("val waiterOptions = StandardRetryStrategyOptions(maxTimeMs = 300_000, maxAttempts = 20)")
```

Comment:
I made them up since the Smithy waiters spec is mum on sensible defaults for maximum time and maximum attempts. These are the kinds of values I'd expect to be user-overridable per-call once we determine how we want per-call configuration to work.